cockpit_api/src/main/java/com/cockpit/api/service/jiragateway/JiraGatewayService.java

```diff
@@ -304,4 +173,201 @@ public void setTotalNbOfUserStoryForEachSprintOfEachProject() {
             }
         }
     }
+
+    @Scheduled(initialDelay = 10 * ONE_SECOND, fixedDelay = 10 * ONE_MINUTE)
+    public void deleteJiraProjects() throws UnirestException {
+        List<Jira> jiraProjectList = jiraRepository.findAllByOrderById();
+        for(Jira jira : jiraProjectList){
+            HttpResponse<JsonNode> jiraProject = Unirest.get(jiraUrl+"/rest/api/3/project/"+jira.getJiraProjectKey())
+                    .basicAuth(username, token)
+                    .header(HEADERKEY, HEADERVALUE)
+                    .asJson();
+            if (jiraProject.getStatus() == 404) {
+                jiraRepository.delete(jira);
+            }
+        }
+    }
+
+    @Scheduled(initialDelay = 10 * ONE_SECOND, fixedDelay = 10 * ONE_MINUTE)
+    public void deleteJiraSprint() throws UnirestException{
+        List<Sprint> sprintList = sprintRepository.findAll();
+        for (Sprint sprint : sprintList){
+            HttpResponse<JsonNode> foundSprint = Unirest.get(jiraUrl+"/rest/agile/1.0/sprint/"+sprint.getJiraSprintId())
+                    .basicAuth(username, token)
+                    .header(HEADERKEY, HEADERVALUE)
+                    .asJson();
+            if (foundSprint.getStatus() == 404) {
+                sprintRepository.delete(sprint);
+            }
+        }
+    }
+
+    @Scheduled(initialDelay = 10 * ONE_SECOND, fixedDelay = 10 * ONE_MINUTE)
+    public void deleteJiraIssues() throws UnirestException {
+        List<UserStory> userStories = userStoryRepository.findAll();
```

Comment:
Similar with delete sprints,
Woud it be better to use this url to get all user stories instead of calling 1000 times the jira server in a for loop?
https://tdf.atlassian.net/rest/api/3/search?jql=issuetype=Story